app/src/main/java/com/example/androidtraining/MainViewModel.kt

```diff
@@ -0,0 +1,72 @@
+package com.example.androidtraining
+
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.androidtraining.models.LoginResponse
+import com.example.androidtraining.models.Product
+import com.example.androidtraining.models.UserInfo
+import com.example.androidtraining.networking.TokenProvider
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class MainViewModel @Inject constructor(private val firstRepository: FirstRepository) :
```

Comment:
There should be no such thing as a `Main` viewmodel. Each screen should have its own dedicated viewmodel, positioned directly to it file-wise in the project folder structure.